File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        mp = {
            ')':'(', '}':'{', ']':'['
        }
        st = []

        for c in s:
            if c not in mp:
                st.append(c)
            else:
                if not st or st[-1] != mp[c]:
                    return False
                st.pop()
        
        return len(st) == 0

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stack;
        unordered_set<string> token {"+", "-", "*", "/"};

        for (const auto &toke : tokens) {
            if (token.count(toke)) {
                int a = stack.top();
                stack.pop();
                int b = stack.top();
                stack.pop();
                if (toke == "+") {
                    int result = b + a;
                    stack.push(result);
                }
                else if (toke == "-") {
                    int result = b - a;
                    stack.push(result);
                }
                else if (toke == "*") {
                    int result = b * a;
                    stack.push(result);
                }
                else {
                    int result = b / a;
                    stack.push(result);
                }
            }
            else {
                stack.push(stoi(toke));
            }
        }
        return stack.top();
    }
};
